openadapt/crud.py

```diff
@@ -109,28 +116,107 @@ def insert_recording(recording_data):
 
 
 def get_latest_recording():
-    return (
-        db
-        .query(Recording)
-        .order_by(sa.desc(Recording.timestamp))
-        .limit(1)
-        .first()
-    )
+    return db.query(Recording).order_by(sa.desc(Recording.timestamp)).limit(1).first()
+
+
+def get_recording_by_id(recording_id):
+    return db.query(Recording).filter_by(id=recording_id).first()
+
+
+def export_sql(recording_id):
+    """Export the recording data as SQL statements.
+
+    Args:
+        recording_id (int): The ID of the recording.
+
+    Returns:
+        str: The SQL statements to insert the recording into the output file.
+    """
+    engine = sa.create_engine(config.DB_URL)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    recording = get_recording_by_id(recording_id)
+
+    if recording:
+        sql = f"INSERT INTO recording VALUES ({recording.id}, {recording.timestamp}, {recording.monitor_width}, {recording.monitor_height}, {recording.double_click_interval_seconds}, {recording.double_click_distance_pixels}, '{recording.platform}', '{recording.task_description}')"
+        logger.info(f"Recording with ID {recording_id} exported successfully.")
+    else:
+        logger.info(f"No recording found with ID {recording_id}.")
+
+    return sql
+
+
+def create_db(recording_id, sql):
+    """Create a new database and import the recording data.
 
+    Args:
+        recording_id (int): The ID of the recording.
+        sql (str): The SQL statements to import the recording.
+
+    Returns:
+        tuple: A tuple containing the timestamp and the file path of the new database.
+    """
+    db.close()
+    db_fname = f"recording_{recording_id}.db"
+
+    t = time.time()
+    shutil.copyfile(config.ENV_FILE_PATH, f"{config.ENV_FILE_PATH}-{t}")
```

Comment:
What do you think about adding some logging here? e.g.
```
source_file_path = config.ENV_FILE_PATH
target_file_path = f"{config.ENV_FILE_PATH}-{t}"
logger.info(f"{source_file_path=} {target_file_path=}")
shutil.copyfile(source_file_path, target_file_path)
```

Reply:
resolved in https://github.com/MLDSAI/OpenAdapt/pull/223/commits/9dc1850918e81648b9885dd8a1be836d5e8d95ef